backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -0,0 +1,375 @@
+                setCommandStatus(CommandStatus.FAILED);
+                return;
+            }
+
+            newImage.setVolumeType(getParameters().getPreallocation());
+        }
+
+        getDiskImage().setActive(false);
+        newImage.getImage().setSequenceNumber(imageDao.getMaxSequenceNumber(getDiskImage().getId()) + 1);
+        TransactionSupport.executeInNewTransaction(() -> {
+            addDiskImageToDb(newImage, getCompensationContext(), true);
+            imageDao.update(getDiskImage().getImage());
+
+            return null;
+        });
+    }
+
+    private void updatePhase(ConvertDiskCommandParameters.ConvertDiskPhase phase) {
+        getParameters().setConvertDiskPhase(phase);
+        persistCommandIfNeeded();
+    }
+
+    private VolumeType getVolumeType(DiskImage diskImage) {
+        return getParameters().getPreallocation() == null ? diskImage.getVolumeType() :
+                getParameters().getPreallocation();
+    }
+
+    private VolumeFormat getVolumeFormat(DiskImage diskImage) {
+        return getParameters().getVolumeFormat() == null ? diskImage.getVolumeFormat() :
+                getParameters().getVolumeFormat();
+    }
+
+    private VdsmImageLocationInfo getLocationInfo() {
+        return (VdsmImageLocationInfo) getParameters().getLocationInfo();
+    }
+
+    @Override
+    protected void setActionMessageParameters() {
+        addValidationMessage(EngineMessage.VAR__ACTION__CONVERT_DISK);
+        addValidationMessage(EngineMessage.VAR__TYPE__DISK);
+    }
+
+    @Override
+    protected Map<String, Pair<String, String>> getExclusiveLocks() {
+        return Collections.singletonMap(getDiskImage().getId().toString(),
+                LockMessagesMatchUtil.makeLockingPair(LockingGroup.DISK,
+                        new LockMessage(EngineMessage.ACTION_TYPE_FAILED_DISK_IS_BEING_CONVERTED)
+                                .with("DiskAlias", getDiskImage().getDiskAlias())));
+    }
+
+    @Override
+    protected LockProperties applyLockProperties(LockProperties lockProperties) {
+        return lockProperties.withScope(LockProperties.Scope.Command);
+    }
+
+    @Override
+    public Map<String, String> getJobMessageProperties() {
+        if (jobProperties == null) {
+            jobProperties = super.getJobMessageProperties();
+            jobProperties.put("diskname", getDiskImage().getName());
+            jobProperties.put("allocationpolicy", getParameters().getPreallocation().toString());
+            jobProperties.put("diskformat", getParameters().getVolumeFormat().toString());
+        }
+
+        return super.getJobMessageProperties();
+
+    }
+
+    @Override
+    public CommandCallback getCallback() {
+        return callbackProvider.get();
+    }
+
+    @Override
+    protected void endSuccessfully() {
+        imageDao.remove(getDiskImage().getImageId());
+
+        freeLock();
+        imagesHandler.updateImageStatus(getParameters().getNewVolGuid(), ImageStatus.OK);
```

Comment:
do we need to execute imageDao.remove and imagesHandler.updateImageStatus within the transaction?

Reply:
I'm not sure, specifically if we successfully removed the entry for the old image but failed to update the status to OK, it seems ok to have them complete separately

Reply:
ack